src/Chapter2.hs

```diff
@@ -336,7 +348,8 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList a b list = if a<0 || b<0 then [] 
+  else take (b-a+1) (drop a list)
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

Reply:
I added check for end>start case. Also refactored function using guards and $ application.